### PR TITLE
Implement the exceptional Jordan algebra

### LINKED ISSUE #32940: Implement octonions and split octonions
<div id="comment:0"></div>

CC:  @slel

Component: **algebra**

Keywords: **octonions**

_Issue created by migration from https://trac.sagemath.org/ticket/32940_

### PR DESCRIPTION
**Pull request by tscrim**
### :books: Description

"The" exceptional Jordan algebra is a 27 dimensional algebra defined as the $3 \times 3$ self-adjoint matrices over the octonions. This is an important Jordan algebra as it is used to construct the simple Lie group/algebra of type $F_4$.

Fixes #32940.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### :hourglass: Dependencies

- #35630 used in the construction

### CODE REVIEW

File: src/sage/algebras/jordan_algebra.py [hunk 1]
```diff
@@ -1052,3 +1063,687 @@ def bar(self):
...
+                True
+            """
+            ret = {}
+            for i in range(3):
+                if self._data[i]:
+                    ret[i] = self._data[i]
+                mc = self._data[3+i].monomial_coefficients()
+                for k, coeff in mc.items():
+                    ret[3+i*8+k] = coeff
+            return ret
```
Comment (mkoeppe) [inline src/sage/algebras/jordan_algebra.py]: Your files are missing LF before EOF

Comment (tscrim) [reply src/sage/algebras/jordan_algebra.py]: Should be fixed now.

**Comment by fchapoton** [comment]
- missing the word "matrix" in the sentence `self-adjoint 3x3  over an octonion`

- do we have a category and a testsuite that checks the Jordan axioms ?

**Comment by tscrim** [comment]
> * missing the word "matrix" in the sentence `self-adjoint 3x3  over an octonion`
Fixed.

>     * do we have a category and a testsuite that checks the Jordan axioms ?

There is no category as I feel the implementations are not complicated enough to warrant one (i.e., an ABC is sufficient). However, there was no `_test_jordan_relations()` method; I have added it. Good idea!

**Comment by tscrim** [comment]
The linter is now happy (to the extent it can be).

**Review by fchapoton — APPROVED** [review]
ok, thanks. Let's move on

**Pull request merged**